src/bokeh/embed/bundle.py

```diff
@@ -159,12 +159,12 @@ def bundle_for_objs_and_resources(objs: Sequence[Model | Document] | None,
         Bundle
 
     '''
-    # Any env vars will overide a local default passed in
-    resources = settings.resources(default=resources)
-    if isinstance(resources, str):
-        resources = Resources(mode=resources)
 
-    if resources is None or isinstance(resources, BaseResources):
+    # TODO: None means no resources, not default
+    if resources is None:
+        resources = Resources(mode=settings.resources())
+
+    if isinstance(resources, BaseResources):
         js_resources = css_resources = resources
     elif isinstance(resources, tuple) and len(resources) == 2 and all(r is None or isinstance(r, BaseResources) for r in resources):
```

Comment:
We could do a git search to try to see if anyone anywhere is utilizing this complexity. I really don't expect anyone is even using `bundle_for_objs_and_resources` since we don't actually include it in the documentation at all. I would very much support drastically streamlining these codepaths.